osi_vehicle.proto

```diff
@@ -0,0 +1,829 @@
+
+            // This is the speed the function targets.
+            // E.g.: At the point of activation, the actual speed could be 80 km/h, 
+            // but the function tries to accelerate to 130 km/h.
+            //
+            // Unit: [km/h]
+            //
+            optional double targeted_speed = 2;
+
+            // The timegap describes the minimumdistance to the next vehicle in front.
+            //
+            // Unit: [s]
+            //
+            optional double timegap = 3;
+            
+            // Possibility to inform about an error, warning or just to give an information.
+            //
+            // \note It is implementation-specific which information should be sent.
+            //
+            optional Notification notification = 4;
+
+            // Definition of possible function-states.
+            //
+            enum FunctionState
+            {
+                // The function is not working.
+                //
+                FUNCTION_STATE_OFF = 0;
+
+                // The function is initializing.
+                // Transition: STATE_OFF ==> STATE_UNAVAILABLE or STATE_AVAILABLE.
+                //
+                FUNCTION_STATE_INITIALIZING = 1;
+                
+                // The function is initialized but not ready to start.
+                //
+                FUNCTION_STATE_UNAVAILABLE = 2;
+
+                // The function is initialized and ready to start.
+                //
+                FUNCTION_STATE_AVAILABLE = 3;
+
+                // The function is starting.
+                // Transition: STATE_AVAILABLE ==> STATE_RUNNING.
+                //
+                FUNCTION_STATE_STARTING = 4;
+
+                // The function is running.
+                //
+                FUNCTION_STATE_RUNNING = 5;
+
+                // The function is stopping.
+                // Transition: STATE_RUNNING ==> STATE_UNAVAILABLE or STATE_AVAILABLE.
+                //
+                FUNCTION_STATE_STOPPING = 6;
+
+                // An Error occured in the function.
+                //
+                FUNCTION_STATE_FAILURE = 7;
+            }
+        }
+
+        //
+        // \brief Interface to describe the communication of an automated-driving function.
+        //
+        // Intentions, requests or wishes from the function to change the vehicle-state.
+        //
+        message Intentions
+        {
+            // All information about the trajectory the vehicle should follow.
+            //
+            optional Trajectory trajectory = 1;
+
+            // Angle, angle-speed and torque.
+            //
+            optional SteeringWheel steering_wheel = 2;
+
+            // Factor to scale the steeringtorque of the function output.
+            //
+            // \note 0 = no force of the function, 0.5 = half the force, 1 = 100% torque.
+            //
+            // Range: [0, 1] 
+            //
+            optional double steering_override_factor = 3;
+
```

Comment:
Maybe pedal (accelerator and/or brake) override here also?

Reply:
Values I would say on the "waiting-list". Is there a use-case?